W-17768108 Einstein AI Connector (#2787)
mulesoft/docs-connectors: einstein-ai/1.0/modules/ROOT/pages/einstein-ai-connector-studio.adoc

> = Using Anypoint Studio to Configure MuleSoft Chain AI Connector 1.0 - Mule 4

Reviewer: ```suggestion
= Using Anypoint Studio to Configure Einstein AI Connector 1.0 - Mule 4
```

> :exchange-search-name: mulesoft ai chain

Reviewer: ```suggestion
:exchange-search-name: einstein ai
```

:connector-name: Einstein AI
:lc-connector-name: einstein-ai
:studio-connector-name: Einstein AI Connector

include::connectors::partial$t-studio-partial.adoc[tags=topic-intro]

[[create-mule-project]]
== Create a Mule Project

include::connectors::partial$t-studio-partial.adoc[tags=create-project-intro]

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

Add Einstein AI Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location and add the required dependencies to the project's `pom.xml` file:

. In *Mule Palette*, click *(X) Search in Exchange*.

> . In *Add Dependencies to Project*, type `mulesoft ai chain` in the search field.

Reviewer: ```suggestion
. In *Add Dependencies to Project*, type `einstein ai` in the search field.
```

. Click *Einstein AI Connector* in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

[[configure-source]]
== Configure a Source

include::connectors::partial$t-studio-partial.adoc[tags=configure-source-intro]

> You can configure one of these sources to use with MuleSoft Chain AI Connector:

Reviewer: ```suggestion
You can configure one of these sources to use with Einstein AI Connector:
```

include::connectors::partial$t-studio-partial.adoc[tags=listener-scheduler-descriptions]

include::connectors::partial$t-studio-partial.adoc[tags=listener-steps]

[[add-connector-operation]]
== Add a Connector Operation to the Flow

When you add a connector operation to your flow, you are specifying an action for that connector to perform.

To add an operation for Einstein AI Connector, follow these steps:

. In *Mule Palette*, select *Einstein AI Connector* and then select the operation to add.
+

> image:einstein-ai-palette-operations.png["Mule palette in Studio with Einstein AI selected and all of the connector's operations listed"]

Reviewer: this image doesn't render

To configure the global element for Einstein AI Connector, follow these steps:

. Select the operation in the Studio canvas.
. On the properties screen for the operation, click the Add (*+*) icon  to access the global element configuration fields.
. In the *Global Elements Properties > General* tab, choose your preferred large language model (LLM) type from *LLM*:
+
** Anthropic
** Azure OpenAI
** Mistral AI
** Ollama
** OpenAI
** GroqAI
. From *Config type*, choose from these configuration types: 
* *Environment Variables*
+
This configuration requires you to set the environment variables in the operating system where Mule runtime is deployed. When you choose this option, enter a `-` in the *File path* field.
+
Based on the LLM type you chose, you must set different environment variables. These are the supported environment variables based on the supported LLM types:
+
** Anthropic: `ANTHROPIC_API_KEY` 
** AWS Bedrock: `AWS_ACCESS_KEY_ID- AWS_SECRET_ACCESS_KEY` 
** Azure OpenAI: `AZURE_OPENAI_ENDPOINT- AZURE_OPENAI_KEY- AZURE_OPENAI_DEPLOYMENT_NAME` 
** MistralAI: `MISTRAL_AI_API_KEY` 
** OpenAI: `OPENAI_API_KEY` 
** Ollama: `OLLAMA_BASE_URL` 
** Groq AI: `GROQ_API_KEY`
+

> image:einstein-ai-config-studio.png["Global Element Properties window with configuration fields filled out for the Environment Variables configuration type"]

Reviewer: image doesnt render

* *Configuration JSON* 
+
This configuration requires you to provide a configuration JSON file with all the required LLM properties.
+
This is an example of the configuration JSON file:
+
[source,json]
----
{
    "OPENAI": {
        "OPENAI_API_KEY": "YOUR_OPENAI_API_KEY"
    },
    "MISTRAL_AI": {
        "MISTRAL_AI_API_KEY": "YOUR_MISTRAL_AI_API_KEY"
    },
    "OLLAMA": {
        "OLLAMA_BASE_URL": "http://baseurl.ollama.com"
    },
    "GROQAI_OPENAI": {
        "GROQ_API_KEY": "YOUR_GROQAI_APIKEY"
    },
    "ANTHROPIC": {
        "ANTHROPIC_API_KEY": "YOUR_ANTHROPIC_API_KEY"
    },
    "AZURE_OPENAI": {
        "AZURE_OPENAI_KEY": "YOUR_AZURE_OPENAI_KEY",
        "AZURE_OPENAI_ENDPOINT": "http://endpoint.azure.com",
        "AZURE_OPENAI_DEPLOYMENT_NAME": "YOUR_DEPLOYMENT_NAME"
    }
}
----
+

> Make sure you fill out the required properties for your LLM type. You can store the file externally or add it directly in the Mule application under the `src/main/resource` folder by using this Dataweave expression:

Reviewer: ```suggestion
Make sure you fill out the required properties for your LLM type. You can store the file externally or add it directly in the Mule application under the `src/main/resource` folder by using this DataWeave expression:
```
